### DISCUSSION TITLE
Authentication Methods

### DISCUSSION THREAD

**Discussion by gbasso23** [microsoftgraph/msgraph-sdk-powershell#2031] [Q&A]
Which Authentication Methods can I use to connect-MSGraph?
So my Problem is, I need a Powershell Script to run on a Server, and it should run every month. Once I login with the normal way, so with a username and password, every time I restart the PowerShell ISE I have to Log back in, which is normal. But I need a way to Login without Userinteraction.
Is there any way I can login into MSGraph without a user interaction?

(Sorry for my bad English)

**Comment by vandre** [comment]
@gbasso23  You can use a managed identity to authenticate to MsGraph without user interaction. See this related [discussion](https://github.com/microsoftgraph/msgraph-sdk-powershell/discussions/1965)
